src/System.Text.Json/src/System/Text/Json/JsonDocument.Parse.cs

```diff
@@ -0,0 +1,298 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Text.Json
+{
+    public sealed partial class JsonDocument
+    {
+        private const int UnseekableStreamInitialRentSize = 4096;
+
+        public static JsonDocument Parse(ReadOnlyMemory<byte> utf8Json, JsonReaderOptions readerOptions = default)
+        {
+            CheckSupportedOptions(readerOptions);
+
+            return Parse(utf8Json, readerOptions, null);
+        }
+
+        public static JsonDocument Parse(ReadOnlySequence<byte> utf8Json, JsonReaderOptions readerOptions = default)
+        {
+            CheckSupportedOptions(readerOptions);
+
+            if (utf8Json.IsSingleSegment)
+            {
+                return Parse(utf8Json.First, readerOptions, null);
+            }
+
+            int length = checked((int)utf8Json.Length);
+            byte[] utf8Bytes = ArrayPool<byte>.Shared.Rent(length);
+
+            try
+            {
+                utf8Json.CopyTo(utf8Bytes.AsSpan());
+                return Parse(utf8Bytes.AsMemory(0, length), readerOptions, utf8Bytes);
+            }
+            catch
+            {
+                // Holds document content, clear it before returning it.
+                utf8Bytes.AsSpan(0, length).Clear();
+                ArrayPool<byte>.Shared.Return(utf8Bytes);
+                throw;
+            }
+        }
+
+        public static JsonDocument Parse(Stream utf8Json, JsonReaderOptions readerOptions = default)
```

Comment:
Also, should the parameter name have stream in it? `utf8JsonStream`?

Reply:
> should the parameter name have stream in it? utf8JsonStream?

The `stream`ness is conveyed in the type.  The `utf8` and `json` are using the name to convey contents expectations.